nvme-print-json.c

```diff
@@ -2121,25 +2121,47 @@ static void json_boot_part_log(void *bp_log, const char *devname,
 
 /* Printable Eye string is allocated and returned, caller must free */
 static char *json_eom_printable_eye(struct nvme_eom_lane_desc *lane,
-				    struct json_object *r)
+									struct json_object *r)
 {
 	char *eye = (char *)lane->eye_desc;
-	char *printable = malloc(lane->nrows * lane->ncols + lane->ncols);
+	uint16_t nrows = lane->nrows;
+	uint16_t ncols = lane->ncols;
+
+	if (nrows == 0 || ncols == 0)
+		return NULL;
+
+	// Allocate buffer for full printable string (with newlines)
+	char *printable = malloc(nrows * ncols + nrows + 1); // +1 for null terminator
```

Comment:
Variable declaration are at the beginning of the block. There only a few exceptions allowed.